### PR TITLE
[QC-871] Add log rotate parameters and refactor how we pass the discard file options 

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

File: Framework/include/QualityControl/QcInfoLogger.h [hunk 1]
```diff
@@ -27,6 +27,14 @@ typedef AliceO2::InfoLogger::InfoLoggerContext infoContext;
...
 {
 
+struct DiscardFileParameters {
+  bool debug = false;
+  int fromLevel = 21 /* Discard Trace */;
+  std::string discardFile;
+  unsigned long rotateMaxBytes = 0;
+  unsigned int rotateMaxFiles = 0;
+};
+
```
Comment (knopers8) [inline Framework/include/QualityControl/QcInfoLogger.h:37]: as a future improvement, I would propose to move this structure to a separate file, so we do not include QcInfologger header in configuration structures.

File: Framework/src/QcInfoLogger.cxx [hunk 2]
```diff
@@ -97,11 +96,12 @@ void QcInfoLogger::init(const std::string& facility,
...
+  DiscardFileParameters discardFileParameters;
   std::string discardDebugStr = config.get<std::string>("qc.config.infologger.filterDiscardDebug", "false");
-  bool discardDebug = discardDebugStr == "true";
-  int discardLevel = config.get<int>("qc.config.infologger.filterDiscardLevel", 21 /* Discard Trace */);
-  std::string discardToFile = config.get<std::string>("qc.config.infologger.filterDiscardFile", "");
-  init(facility, discardDebug, discardLevel, discardToFile, dplInfoLogger, dplContext, run, partitionName);
+  discardFileParameters.debug = discardDebugStr == "true";
+  discardFileParameters.fromLevel = config.get<int>("qc.config.infologger.filterDiscardLevel", 21 /* Discard Trace */);
+  discardFileParameters.discardFile = config.get<std::string>("qc.config.infologger.filterDiscardFile", "");
+  init(facility, discardFileParameters, dplInfoLogger, dplContext, run, partitionName);
```
Comment (knopers8) [inline Framework/src/QcInfoLogger.cxx:104]: why are you not reloading all the parameters?

**Review by knopers8 — APPROVED** [review]

File: doc/Advanced.md [hunk 3]
```diff
@@ -935,8 +935,10 @@ should not be present in real configuration files.
         "filterDiscardDebug": "false",    "": "Set to 1 to discard debug and trace messages (default: false)",
         "filterDiscardLevel": "2",        "": "Message at this level or above are discarded (default: 21 - Trace)",
         "filterDiscardFile": "",          "": ["If set, the discarded messages will go to this file (default: <none>)",
-                                              "The keyword _ID_, if used, is replaced by the device ID.", 
-                                              "Discarded Debug messages don't go there."]
+                                              "The keyword _ID_, if used, is replaced by the device ID.",
+                                              "Discarded Debug messages don't go there."],
+        "filterRotateMaxBytes": "",       "": "Maximum size of the discard file.", 
+        "filterRotateMaxFiles": "",       "": "Maximum number of discard files."
```
Comment (knopers8) [inline doc/Advanced.md:941]: why there could be more than one?

File: Framework/include/QualityControl/QcInfoLogger.h (hunk 1 above)
Comment (Barthelemy) [reply Framework/include/QualityControl/QcInfoLogger.h:37]: agreed, will do

File: doc/Advanced.md (hunk 3 above)
Comment (Barthelemy) [reply doc/Advanced.md:941]: isn't that how logrotate usually works ? when a file is full it creates a new one, within the limit of the number of files ?

File: Framework/src/QcInfoLogger.cxx (hunk 2 above)
Comment (Barthelemy) [reply Framework/src/QcInfoLogger.cxx:104]: good point, will fix.

File: doc/Advanced.md (hunk 3 above)
Comment (knopers8) [reply doc/Advanced.md:941]: ah yes, of course

**Comment by Barthelemy** [comment]
as o2-cs8 and o2-dataflow-cs8 have passed, I merged to be able to release

**Pull request merged**